### PR TITLE
Add developer tools guides

### PR DESCRIPTION
**Pull request by eballantine**
Not sure if there's more to say here?

### CODE REVIEW

File: docs/developer_tools/canvas_lint.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+The `canvas lint` command will run the following validation checks:
+
+1. RequiredFilesCheck
+    1. Checks that the required files exist (e.g. `templates/product/index.html`).
+2. ValidHtmlCheck
+    1. Validates all `.html` files contain valid HTML.
+3. ValidLiquidCheck
+    1. Validates all `.html` files contain valid Liquid.
+4. COMING SOON: Validate the schemas in blocks, headers and footers.
```
Comment (paulina-a) [inline docs/developer_tools/canvas_lint.md:18]: I may be wrong but I _think_ the coming soons are now here:

- RequiredFiles
- ValidHtml
- ValidLiquid
- ValidJson
- ValidSass
- ValidBlockSchemas
- ValidMenuSchema
- ValidFooterSchema
- ValidCustomTypes

**Review by paulina-a — APPROVED** [review]
Looks good, we can always build on it later

**Review by paulina-a — APPROVED** [review]
Looks good - I don't know the exact explanation of each of the checks though so if you haven't already it may be good to get a member of 945 team to review

**Comment by eballantine** [comment]
> Looks good - I don't know the exact explanation of each of the checks though so if you haven't already it may be good to get a member of 945 team to review

I checked the code and there's very clear tests for how it should behave and all the error messages that could/would be shown!

**Pull request merged**